templates/pod-monitor.yml

```
@@ -0,0 +1,17 @@
+---
+apiVersion: monitoring.coreos.com/v1
+kind: PodMonitor
+metadata:
+  name:  {{ template "app.name" . }}-pod-monitor
+spec:
+  selector:
+    matchLabels:
+      app: {{ template "app.name" . }}
+  podMetricsEndpoints:
+    - path: metrics
+      scheme: http
+      scrapeTimeout: 5s
+      port: http
+  namespaceSelector:
+    matchNames:
+      - {{ .Release.Namespace }}
```

We should have this also be enabled only when the metrics value is passed:
```suggestion
{{- if .Values.metrics.enabled }}
---
apiVersion: monitoring.coreos.com/v1
kind: PodMonitor
metadata:
  name:  {{ template "app.name" . }}-pod-monitor
spec:
  selector:
    matchLabels:
      app: {{ template "app.name" . }}
  podMetricsEndpoints:
    - path: metrics
      scheme: http
      scrapeTimeout: 5s
      port: http
  namespaceSelector:
    matchNames:
      - {{ .Release.Namespace }}
{{- end }}

```

---

good catch @pereztr5